test/unit/modules/delegates.js

```diff
@@ -489,6 +490,172 @@ describe('delegates', () => {
 				});
 			});
 		});
+
+		describe('getDelegateKeypairForCurrentSlot', () => {
+			let delegates;
+			let __private;
+
+			before(done => {
+				delegates = library.rewiredModules.delegates.__get__('self');
+				__private = library.rewiredModules.delegates.__get__('__private');
+
+				__private.keypairs = {
+					// genesis_1 delegate keypairs
+					'9d3058175acab969f41ad9b86f7a2926c74258670fe56b37c429c01fca9f2f0f': {
+						publicKey: new Buffer.from(
+							'9d3058175acab969f41ad9b86f7a2926c74258670fe56b37c429c01fca9f2f0f',
+							'hex'
+						),
+						privateKey: new Buffer.from(
+							'b092a6664e9eed658ff50fe796ee695b9fe5617e311e9e8a34eb340eb5b831549d3058175acab969f41ad9b86f7a2926c74258670fe56b37c429c01fca9f2f0f',
+							'hex'
+						),
+					},
+					// genesis_2 delegate keypairs
+					'141b16ac8d5bd150f16b1caa08f689057ca4c4434445e56661831f4e671b7c0a': {
+						publicKey: new Buffer.from(
+							'141b16ac8d5bd150f16b1caa08f689057ca4c4434445e56661831f4e671b7c0a',
+							'hex'
+						),
+						privateKey: new Buffer.from(
+							'e896941f49df461e34d87cbe66c9cf35abcc58a25e4494c97a3f781b9e77e83c141b16ac8d5bd150f16b1caa08f689057ca4c4434445e56661831f4e671b7c0a',
+							'hex'
+						),
+					},
+					// genesis_3 delegate keypairs
+					'3ff32442bb6da7d60c1b7752b24e6467813c9b698e0f278d48c43580da972135': {
+						publicKey: new Buffer.from(
+							'3ff32442bb6da7d60c1b7752b24e6467813c9b698e0f278d48c43580da972135',
+							'hex'
+						),
+						privateKey: new Buffer.from(
+							'd4b9f54978f789c1a291f948d32f19039d54962b5b4c14d314c1521f2975b9be3ff32442bb6da7d60c1b7752b24e6467813c9b698e0f278d48c43580da972135',
+							'hex'
+						),
+					},
+				};
```

Comment:
I feel non-confident of mentioning private keys. In our config.json we have delegates in the form; 

```
{
   "encryptedPassphrase":
          "iterations=1&salt=476d4299531718af8c88156aab0bb7d6&cipherText=663dde611776d87029ec188dc616d96d813ecabcef62ed0ad05ffe30528f5462c8d499db943ba2ded55c3b7c506815d8db1c2d4c35121e1d27e740dc41f6c405ce8ab8e3120b23f546d8b35823a30639&iv=1a83940b72adc57ec060a648&tag=b5b1e6c6e225c428a4473735bc8f1fc9&version=1",
    "publicKey":
          "9d3058175acab969f41ad9b86f7a2926c74258670fe56b37c429c01fca9f2f0f"
},
```

I saw you used same public keys, so I suggest to load their encrypted pass phrase from config runtime and generate the private keys, don't hardcode private keys.